mvc/overview.rst

```diff
@@ -1,8 +1,224 @@
+
+.. code-block:: html
+
+  <ul>
+    @for (int i = 0; i < 5; i++) {
+      <li>List item @i</li>
+    }
+  </ul>
+
+Using the MVC Razor view engine you can define layouts, partial views and replaceable sections.
+
+Strongly typed views
+--------------------
+
+Razor views in MVC can be strongly typed based on your model. A strongly-typed view explicitly specifies the model type for the view. You can then leverage the compiler and IntelliSense to ensure that references to property names and types in your views are correct:
+
+.. code-block:: html
+
+  @model IEnumerable<MvcMusicStore.Models.Album>
+  <ul> 
+      @foreach (Album p in Model) 
+      { 
+          <li>@p.Title</li>
+      }
+  </ul>
+
+HTML Helpers
+------------
+
+HTML Helpers are .NET methods that you can use in Razor views to generate HTML. HTML Helpers help you generate views based on your model while handling concerns like HTML encoding for you. 
+
+.. code-block:: html
+  :emphasize-lines: 1
+
+  @using (Html.BeginForm("Search", "Home", FormMethod.Get)) 
+  {
+      <input type="text" name="q" />
+      <input type="submit" value="Search" />
+  }
+
+MVC includes a comprehensive set of HTML Helpers for generating forms, labels, input elements, validation messages and links. You can customize the behavior of HTML Helpers for specific types using editor and display templates.
+
+Tag Helpers
+-----------
+:doc:`Tag Helpers </views/tag-helpers/intro>` enable you to dynamically generate or modify the HTML returned to the client. You can use tag helpers to define custom tags or to modify the behavior existing tags. Tag Helpers bind to specific elements based on the element name and its attributes. Tag Helpers give you all of the benefits of server-side rendering while still preserving an HTML editing experience.
+
+For example, you can use built-in ``LinkTagHelper`` to create a link to the ``Login`` action of your ``AccountsController`` like this:
+
+.. code-block:: html
+  :emphasize-lines: 3
+  
+  <p>
+      Thank you for confirming your email. 
+      Please <a asp-controller="Account" asp-action="Login">Click here to Log in</a>.
+  </p>
+
+Or you can use the ``EnvironmentTagHelper`` to include different scripts in your views (ex full vs minified) based on the runtime environment (i.e. Development vs Production):
+
+.. code-block:: html
+
+  <environment names="Development">
+      <script src="~/lib/jquery/dist/jquery.js"></script>
+  </environment>
+  <environment names="Staging,Production">
+      <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.1.4.min.js"
+              asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
+              asp-fallback-test="window.jQuery">
+      </script>
+  </environment>
+
+MVC includes a full set of built-in Tag Helpers that you can use or your can build your own.
```

Comment:
For HTML helpers the term "comprehensive set" was used - that's perhaps a better fit.